src/content/en/tools/workbox/guides/migrate.md

```diff
@@ -0,0 +1,224 @@
+book_path: /web/tools/workbox/_book.yaml
+description: A guide to migrating from sw-precache or sw-toolbox to Workbox.
+
+{# wf_updated_on: 2018-01-25 #}
+{# wf_published_on: 2018-01-25 #}
+{# wf_blink_components: N/A #}
+
+# Migrate from sw-precache or sw-toolbox {: .page-title }
+
+{% include "web/tools/workbox/_shared/alpha.html" %}
+
+Developers who have previously used [`sw-precache`](https://github.com/GoogleChromeLabs/sw-precache)
+and/or [`sw-toolbox`](https://github.com/GoogleChromeLabs/sw-toolbox) have a straightforward upgrade
+path to the Workbox family of libraries. Upgrading to Workbox will provide a modern, extensible
+service worker experience with improved debugging and developer ergonomics.
+
+## Modifications to your existing configuration
+
+Regardless of which specific build tools you're migrating from, keep in mind the following name
+changes and deprecations when putting together your equivalent Workbox configuration.
+
+### Renamed options
+
+- The `runtimeCaching` config parameter has been
+[updated](https://github.com/GoogleChrome/workbox/issues/1096#issue-278581349) to take a more
+streamlined set of options, corresponding to the underlying Workbox caching plugins.
+
+- Within `runtimeCaching`, the `fastest` handler name has been
+[renamed](https://github.com/GoogleChrome/workbox/issues/915) to `staleWhileRevalidate`, which
+provides equivalent functionality.
+
+- The `dynamicUrlToDependencies` config parameter has been renamed `templatedUrls`.
+
+- The `staticFileGlobs` config parameter has been renamed `globPatterns`.
+
+### Deprecated options
+
+- Express-style wildcard routes are [no longer
+supported](https://github.com/GoogleChrome/workbox/issues/1012). If you were using Express-style
+wildcard routes in either the `runtimeCaching` configuration or directly in `sw-toolbox`, please
+migrate to an equivalent regular expression route when using Workbox.
+
+## sw-precache migrations
+
+### From the sw-precache CLI to workbox-cli
+
+Developers using the `sw-precache` command line interface, either running the command manually or as
+part of an [`npm scripts`](https://docs.npmjs.com/misc/scripts)-based build process, will find using
+the the [`workbox-cli` module](http://npmjs.com/package/workbox-cli) to be the easiest way to
+migrate. Installing `workbox-cli` will give you access to a binary called `workbox`.
+
+While the `sw-precache` CLI
+[supported](https://github.com/GoogleChromeLabs/sw-precache#command-line-interface) configuring via
+either command line flags or a configuration file, the `workbox` CLI requires that all configuration
+options be provided in a configuration file, using [CommonJS
+`module.exports`](https://nodejs.org/api/modules.html#modules_module_exports).
+
+The `workbox` CLI supports a number of different modes (use `workbox --help` to see all of them),
+but the mode that most closely matches `sw-precache`'s functionality is `generateSW`. So a call to
```

Comment:
This is the start of a new sentence. This is one of the rare cases where its acceptable to start a sentence with 'but'. That explanation's longer than the previous one.